src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1211,6 +1214,12 @@ private String getBranchName(Branch branch)
         return name;
     }
 
+    private String getBranchNameShort(Branch branch)
+    {
+        String name = branch.getName();
+        return name.substring(name.lastIndexOf('/')+1);
```

Comment:
If the goal is to create a name that can be used as a local branch name that can be used with maven release:prepare commits to the branch, then the substring should only remove the "origin/" prefix.  Maven release:prepare will git push origin localBranch:localBranch (uses same branch name for local and remote branch) so the local name needs to match the remote branch name sans "origin/"